### ISSUE TITLE
Add support for Umbraco Deploy

### ISSUE THREAD

**Issue by robertjf** [patrickdemooij9/SeoToolkit.Umbraco#232]
To be truly useful on an Umbraco Cloud instance, we need an Umbraco Deploy plugin to be able to transfer both the SEO Node data and the configuration across from one instance to another.

To support this, we will need to modify the database tables and any reference to the integer-based NodeId to use the Guid-based Key instead, as there will be conflicts/inconsistencies in the integer-based NodeIds from one instance to another.

Currently, if we implement SEO Toolkit on an Umbraco Cloud project, we need to copy over any configuration as well as any metadata if we restore or push nodes between instances.

**Comment by patrickdemooij9** [comment]
Good point. I really need to get back to https://github.com/patrickdemooij9/SeoToolkit.Umbraco/pull/175. That also probably needs to have the same changes done as you suggest here.

**Comment by robertjf** [comment]
Possibly one of the biggest hurdles is the fact that content is stored in the database - Matt Brailesford may have an approach to this that could work - see https://github.com/vendrhub/vendr-deploy - there's an issue (https://github.com/vendrhub/vendr-deploy/issues/1 - Deploy Stock Values) that talks about data in databases but is unresolved, so not sure if he got it working.

How hard would it be to reconfigure SEO Toolkit to store data in a property on the document itself instead?  Even if we somehow "hid" that property from the standard editor and placed it in the Content App where it currently is...  Of course, the configuration will still need to be deployed (which still exists in database)

**Comment by aguyfromdenmark** [comment]
To us, this is pretty much a must have feature. So +1 on this.

**Comment by timeverts** [comment]
@patrickdemooij9, it's been a couple years since there was any feedback on this issue. Is there any update with this?

**Comment by patrickdemooij9** [comment]
Hi @timeverts 

I do actually have an update for this after working on the uSync addon. I tried to also create a deploy package (and the branch is still out there) however there is a fundament issue with deploy with makes it so that SeoToolkit doesn't work with it. 

For uSync, we are able to automatically sync after a content/documenttype is saved/published. However this isn't possible with Deploy. Deploy works with the assumption that everything lives in content trees, SeoToolkit doesn't have that. I've looked through the codebase, but there is no way for me to hook into it with the current setup of SeoToolkit. I also created an issue on the forums: https://forum.umbraco.com/t/custom-content-dependencies-for-umbraco-deploy/6990 and they mention it currently isn't possible.

Until that is picked up, I can't add support for Deploy yet. I really want to, but I am limited by the fact that it isn't possible and Deploy isn't open source so I also can't help with building the capability.

**Comment by robertjf** [comment]
@patrickdemooij9 seems the blocker for Deploy has been resolved since those discussions - the new code uses the new Ridealong feature.  I had claude review the comments and has confirmed, although found a few issues in the current PR - these have been resolved, but introduce further changes into the `SeoToolkit.Umbraco.MetaFields` to support exporting to disk (supports auto-deploy-on-save and exporting the node to disk with it's dependencies) and deleting the on-disk file.
